pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1525,6 +1548,28 @@ func (d *awsDisk) describeVolume() (*ec2.Volume, error) {
 	return volumes[0], nil
 }
 
+// applyUnSchedulableTaint applies a unschedulable taint to a node after verifying
+// if node has become unusable because of volumes getting stuck in attaching state.
+func (c *Cloud) applyUnSchedulableTaint(nodeName types.NodeName, reason string) {
+	node, fetchErr := c.kubeClient.CoreV1().Nodes().Get(string(nodeName), metav1.GetOptions{})
```

Comment:
`AddOrUpdateTaintOnNode` immediately calls [`get`](https://github.com/kubernetes/kubernetes/blob/d6325933e1b5ac8a2c0a5a65ab8bd690015f4697/pkg/controller/controller_utils.go#L905) for the node, so this is causing 2x the calls. You could check for not found error below to accomplish the same log w/ less API calls.

Reply:
We still need the node object below to create event that is emitted in next line.